Zend/zend_language_parser.y

```diff
@@ -362,6 +362,9 @@ top_statement:
 	|	T_USE use_declarations ';'					{ $$ = $2; $$->attr = ZEND_SYMBOL_CLASS; }
 	|	T_USE use_type use_declarations ';'			{ $$ = $3; $$->attr = $2; }
 	|	T_CONST const_list ';'						{ $$ = $2; }
+	|	T_DECLARE const_list ';'
```

Comment:
This should probably be in `statement:` with the other declare definition? Otherwise there will be a weird discrepancy where parens must be used in inner statements.

```php
// Works
if (true) {
    declare(ticks = 1);
}

// Parser error
if (true) {
    declare ticks = 1;
}
```

Reply:
@iluuu1994 Are we sure declare should be allowed inside any other statement?
By design declare has to be the very first statements.

Reply:
>  By design declare has to be the very first statements.

That's not the case for the existing `declare` with parens though. The RFC doesn't mention handling the cases differently thus I think they should behave the same.

Reply:
@iluuu1994 that is indeed true. I'll modify it by introducing `declare_list: '(' const_list ') | const_list;` and referencing it from the `statement`.

Reply:
@iluuu1994 after thinking a while maybe we actually should provide it as a `top_statement` just like `const` is a top level statement and leave the expression syntax as is for now which allow it's depreciation in future along with ticks which have a really narrow use.